biofefi/machine_learning/nn_models.py

```diff
@@ -0,0 +1,271 @@
+            raise ValueError(f"Error occured during prediction of BRNN Classifier: {e}")
+
+
+class BayesianRegularisedNNRegressor(BaseNetwork, BaseEstimator, RegressorMixin):
+    """
+    This class defines a Bayesian Regularized Neural
+    Network for regression tasks.
+
+    Args:
+        problem_type (ProblemTypes): The type of problem
+        (Regression).
+        - layer1: The first linear layer of the network.
+        - layer2: The second linear layer of the network.
+        - output_layer: The output linear layer of the network.
+    """
+
+    def __init__(self, problem_type: ProblemTypes):
+        """
+        Initializes the BayesianRegularisedNNRegressor class.
+        """
+        super().__init__()
+        self.problem_type = problem_type
+        self._name = ModelNames.BRNNRegressor
+        self.layer1, self.layer2, self.output_layer = None, None, None
+
+    def _initialize_network(self, input_dim, output_dim):
+        """
+        Initializes the network layers for BRNN regression.
+
+        Args:
+            input_dim (int): The input dimension of the data.
+            output_dim (int): The output dimension of the
+            data, determined dynamically.
+        """
+        # Define hidden layers and output layer
+        self.layer1 = nn.Linear(input_dim, BrnnOptions.hidden_dim)
+        self.layer2 = nn.Linear(BrnnOptions.hidden_dim, BrnnOptions.hidden_dim)
+        self.output_layer = nn.Linear(BrnnOptions.hidden_dim, output_dim)
+
+        # Initialize weights and optimizer
+        self._initialise_weights()
+        self._get_num_params()
+        self._make_optimizer(OptimiserTypes.Adam, BrnnOptions.lr)
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        """
+        Defines the forward pass of the network.
+
+        Args:
+            x (torch.Tensor): The input data.
+
+        Returns:
+            torch.Tensor: The output after applying the forward
+            pass through the network.
+
+        Raises:
+            ValueError: If an error occurs during the forward pass.
+        """
+        try:
+            x = F.leaky_relu(self.layer1(x), negative_slope=0.01)
+            x = F.leaky_relu(self.layer2(x), negative_slope=0.01)
+            x = self.output_layer(x)
+            return x
+        except Exception as e:
+            raise ValueError(
+                f"Error occured during forward pass of BRNN Regressor: {e}"
+            )
+
+    def fit(self, X: np.ndarray, y: np.ndarray) -> None:
+        """
+        Train the Bayesian Regularized Neural Network.
+
+        Args:
+            X (np.ndarray): The input data.
+            y (np.ndarray): The target data.
+
+        Raises:
+            ValueError: If an error occurs during training.
+        """
+
+        X = torch.tensor(X, dtype=torch.float32)
+        y = torch.tensor(y, dtype=torch.float32).squeeze().long()
+        input_dim = X.shape[1]
+
+        if self.problem_type == ProblemTypes.Regression:
```

Comment:
This is a regressor, you don't need to check the problem type here

Reply:
Ok Dan